libsql/internal/http/http.go

```diff
@@ -90,25 +119,124 @@ func callSqld(ctx context.Context, url string, sql string, sqlParams params) (*r
 		return nil, fmt.Errorf("failed to execute SQL: %s\n%s", sql, err_response.Message)
 	}
 
-	type errObject struct {
-		Message string `json:"message"`
-	}
+	var results []httpResults
 
-	var results []struct {
-		Results *resultSet `json:"results"`
-		Error   *errObject `json:"error"`
-	}
 	if err := json.Unmarshal(body, &results); err != nil {
 		return nil, err
 	}
-	if len(results) != 1 {
-		return nil, fmt.Errorf("wrong number of results for SQL: %s\nexpected 1 got %d", sql, len(results))
-	}
+
 	if results[0].Error != nil {
 		return nil, fmt.Errorf("failed to execute SQL: %s\n%s", sql, results[0].Error.Message)
 	}
 	if results[0].Results == nil {
 		return nil, fmt.Errorf("no results for SQL: %s", sql)
 	}
-	return results[0].Results, nil
+	return results, nil
+}
+
+func convertArgs(args []driver.NamedValue) (parameters, error) {
+	if len(args) == 0 {
+		return parameters{}, nil
+	}
+
+	sortedArgs := [](*driver.NamedValue){}
+	for idx := range args {
+		sortedArgs = append(sortedArgs, &args[idx])
+	}
+	sort.Slice(sortedArgs, func(i, j int) bool {
+		return sortedArgs[i].Ordinal < sortedArgs[j].Ordinal
+	})
+
+	params := parameters{paramsType: getParamType(sortedArgs[0])}
+	for _, arg := range sortedArgs {
+		if params.paramsType != getParamType(arg) {
+			return parameters{}, fmt.Errorf("driver does not accept positional and named parameters at the same time")
+		}
+		params.paramsValues = append(params.paramsValues, struct {
+			name  string
+			value any
+		}{name: arg.Name, value: arg.Value})
+	}
+	return params, nil
+}
+
+func getParamType(arg *driver.NamedValue) paramsType {
```

Comment:
This belongs in http/driver.go

Reply:
I would change this function to `isNamedParam` and drop the enum at the top.

Reply:
already addressed bool vs enum on another comment